tests/test-util/test.manifest.js

```diff
@@ -0,0 +1,105 @@
+          })
+      );
+    });
+
+    it('reports an error for a missing manifest file', () => {
+      let nonExistantFile = '/dev/null/nowhere/manifest.json';
+      return getValidatedManifest(nonExistantFile)
+        .then(makeSureItFails())
+        .catch(onlyInstancesOf(InvalidManifest, (error) => {
+          assert.match(error.message, /Could not read manifest\.json/);
+          // Make sure the filename is included in the exception message.
+          // This is actually done by default in file system error messages.
+          assert.include(error.message, nonExistantFile);
+        }));
+    });
+
+    it('reports an error for invalid manifest JSON', () => withTempDir(
+      (tmpDir) => {
+        let badManifest = 'NOT ... VALID ... JSON';
+        let manifestFile = path.join(tmpDir.path(), 'manifest.json');
+        return fs.writeFile(manifestFile, badManifest)
+          .then(() => getValidatedManifest(manifestFile))
+          .then(makeSureItFails())
+          .catch(onlyInstancesOf(InvalidManifest, (error) => {
+            assert.match(error.message, /Error parsing manifest\.json/);
+            assert.include(error.message, manifestFile);
+          }));
+      }
+    ));
+
+    it('reports an error for a manifest without a name', () => withTempDir(
+      (tmpDir) =>
+        writeManifest(tmpDir.path(), {version: '0.0.1'})
+        .then((manifestFile) => {
+          return getValidatedManifest(manifestFile)
+            .then(makeSureItFails())
+            .catch(onlyInstancesOf(InvalidManifest, (error) => {
+              assert.match(
+                error.message,
+                /Manifest at .* is invalid: missing "name" property/);
+              assert.include(error.message, manifestFile);
+            }));
+        })
+    ));
+
+    it('reports an error for a manifest without a version', () => withTempDir(
+      (tmpDir) =>
+        writeManifest(tmpDir.path(), {name: 'the extension'})
+        .then((manifestFile) => {
+          return getValidatedManifest(manifestFile)
+            .then(makeSureItFails())
+            .catch(onlyInstancesOf(InvalidManifest, (error) => {
+              assert.match(
+                error.message,
+                /Manifest at .* is invalid: missing "version" property/);
+              assert.include(error.message, manifestFile);
+            }));
+        })
+    ));
+
+    it('reports all errors', () => withTempDir(
+      (tmpDir) =>
+        writeManifest(tmpDir.path(), {}) // empty manifest
+        .then((manifestFile) => {
+          return getValidatedManifest(manifestFile)
+            .then(makeSureItFails())
+            .catch(onlyInstancesOf(InvalidManifest, (error) => {
+              assert.match(
+                error.message,
+                /missing "name" property; missing "version" property/);
+            }));
+        })
+    ));
+
+  });
+
+});
+
+
+function writeManifest(destDir, manifestData) {
```

Comment:
The reason I didn't was because I'm not sure any other test will need this helper function. All other tests should be able to work directly with manifest objects.